assets/scripts/app.js

```diff
@@ -0,0 +1,151 @@
+const confirmAddMovieButton = cancelAddMovieButton.nextElementSibling;
+const userInputs = addMovieModal.querySelectorAll('input');
+const entryTextSection = document.getElementById('entry-text');
+const deleteMovieModal = document.getElementById('delete-modal');
+
+const movies = [];
+
+const toggleBackdrop = () => {
+  backdrop.classList.toggle('visible');
+};
+
+const updateUI = () => {
+  if (movies.length === 0) {
+    entryTextSection.style.display = 'block';
+  } else {
+    entryTextSection.style.display = 'none';
+  }
+};
+
+const closeMovieDeletionModal = () => {
+  toggleBackdrop();
+  deleteMovieModal.classList.remove('visible');
+};
+
+const deleteMovieHandler = movieId => {
+  let movieIndex = 0;
+  for (const movie of movies) {
+    if (movie.id === movieId) {
+      break;
+    }
+    movieIndex++;
+  }
+  movies.splice(movieIndex, 1);
+  const listRoot = document.getElementById('movie-list');
+  listRoot.children[movieIndex].remove();
+  closeMovieDeletionModal();
+  updateUI();
+};
+
+
+const startDeleteMovieHandler = movieId => {
+
+  deleteMovieModal.classList.add('visible');
+  toggleBackdrop();
+  const cancelMovieDeletionBtn = deleteMovieModal.querySelector('.btn--passive');
+  let confirmMovieDeletionBtn = deleteMovieModal.querySelector('.btn--danger');
+
+  confirmMovieDeletionBtn.replaceWith(confirmMovieDeletionBtn.cloneNode(true));
+  
+  confirmMovieDeletionBtn = deleteMovieModal.querySelector('.btn--danger');
+
+
+  cancelMovieDeletionBtn.addEventListener('click', closeMovieDeletionModal);
+  
+  confirmMovieDeletionBtn.addEventListener('click', deleteMovieHandler.bind(null, movieId));
+  
+  
+};
+
+const renderNewMovieElement = (id, title, imageUrl, rating) => {
+  const newMovieElement = document.createElement('li');
+  newMovieElement.className = 'movie-element';
+  newMovieElement.innerHTML = `
+    <div class="movie-element__image">
+      <img src="${imageUrl}" alt="${title}">
+    </div>
+    <div class="movie-element__info">
+      <h2>${title}</h2>
+      <p>${rating}/5 stars</p>
+    </div>
+  `;
+  newMovieElement.addEventListener('click', startDeleteMovieHandler.bind(null, id));
+  const listRoot = document.getElementById('movie-list');
+  listRoot.append(newMovieElement);
+};
+
+
+const closeMovieModal = () => {
+  addMovieModal.classList.remove('visible');
+};
+
+const showMovieModal = () => {
+  addMovieModal.classList.add('visible');
+  toggleBackdrop();
+};
+
+const clearMovieInput = () => {
+  for (const usrInput of userInputs) {
+    usrInput.value = '';
+  }
+};
+
+const cancelAddMovieHandler = () => {
+  closeMovieModal();
+  toggleBackdrop();
+  clearMovieInput();
+};
+
+const addMovieHandler = () => {
+  const titleValue = userInputs[0].value;
+  const imageUrlValue = userInputs[1].value;
+  const ratingValue = userInputs[2].value;
+
+  if (
+    titleValue.trim() === '' ||
+    imageUrlValue.trim() === '' ||
+    ratingValue.trim() === '' ||
```

Comment:
Checking for falsey value, can you use shorthand? For line no. 111, 112 and 113

Reply:
fixed @Tanvir04khan